numerical/splines/definitions.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+from numpy import power as Power
+
+from numerical import derivative
+
+
+def linear(x: np.array) -> np.array:
+    """ One-dimensional linear spline.
+
+    Spline defined on the interval (-1, 1).
+    """
+    x = np.array(x, dtype=np.float64)
+    fx = np.zeros_like(x)
+    idx = np.logical_and(x > -1, x <= 0)
+    fx[idx] = x[idx] + 1
+    idx = np.logical_and(x > 0, x < 1)
+    fx[idx] = -x[idx] + 1
+    idx = np.logical_or(x <= -1, x >= 1)
+    fx[idx] = 0
+    return fx
+
+
+@derivative.setup(ftype="numerical")
+def schoenberg(x: np.array) -> np.array:
+    """ One-dimensional Shenberg spline of 5-th order.
+
+    Spline defined on the interval (0, 3)
+    """
```

Comment:
Please add formulas for all cases. You can use Wiki or md file and put a ref on them here.

Reply:
Good idea, but not in this commit, if you don't mind.